### PR TITLE
Run cargo-fmt on all of `datafusion-functions`

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/9281

## Rationale for this change

The way we were defining modules in `datafusion-functions` meant that rustfmt was not running on them 

## What changes are included in this PR?

This applies the fix in https://github.com/apache/arrow-datafusion/pull/9367 to all the packages in the `datafusion-functions` crate

## Are these changes tested?
covered by CI

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/functions/src/lib.rs [hunk 1]
```diff
@@ -90,14 +90,25 @@ pub mod macros;
 pub mod core;
 make_stub_package!(core, "core_expressions");
 
-make_package!(
-    encoding,
-    "encoding_expressions",
-    "Hex and binary `encode` and `decode` functions."
-);
+/// Encoding expressions.
```
Comment (alamb) [inline datafusion/functions/src/lib.rs:93]: This is the actual change of this PR. The rest of the PR is the result of running `cargo fmt` and checking it in

**Review by Jefffrey — APPROVED** [review]
👍

**Comment by alamb** [comment]
I am going to merge this one in as it will cause conflicts (including logical conflicts) as we port the code to datafusion-functions

**Comment by alamb** [comment]
Thanks for the review @Jefffrey

**Pull request merged**